plugins/UFPWriter/UFPWriter.py

```diff
@@ -72,25 +61,23 @@ def write(self, stream, nodes, mode = MeshWriter.OutputMode.BinaryMode):
         gcode.write(gcode_textio.getvalue().encode("UTF-8"))
         archive.addRelation(virtual_path = "/3D/model.gcode", relation_type = "http://schemas.ultimaker.org/package/2018/relationships/gcode")
 
-        # TODO temporarily commented out, as is causes a crash whenever the UFPWriter is called outside of the main thread
-        # self._createSnapshot()
-        #
-        # # Store the thumbnail.
-        # if self._snapshot:
-        #     archive.addContentType(extension = "png", mime_type = "image/png")
-        #     thumbnail = archive.getStream("/Metadata/thumbnail.png")
-        #
-        #     thumbnail_buffer = QBuffer()
-        #     thumbnail_buffer.open(QBuffer.ReadWrite)
-        #     thumbnail_image = self._snapshot
-        #     thumbnail_image.save(thumbnail_buffer, "PNG")
-        #
-        #     thumbnail.write(thumbnail_buffer.data())
-        #     archive.addRelation(virtual_path = "/Metadata/thumbnail.png",
-        #                         relation_type = "http://schemas.openxmlformats.org/package/2006/relationships/metadata/thumbnail",
-        #                         origin = "/3D/model.gcode")
-        # else:
-        #     Logger.log("d", "Thumbnail not created, cannot save it")
+        # Attempt to store the thumbnail, if any:
+        backend = CuraApplication.getInstance().getBackend()
+        snapshot = None if getattr(backend, "getLatestSnapshot", None) is None else backend.getLatestSnapshot()
```

Comment:
Hmm, could consider making this `getLatestSnapshot` function part of `UM.Backend` in the form of a stub so that we don't have to check this.
I think it's fine to leave it for now, but putting it out there if you agree it's better.

Reply:
Actually, thinking about the scope of the class, it's really not supposed to be part of a back-end application so I'm not convinced this should be changed any more.